lib/Sema/CSSimplify.cpp

```diff
@@ -4768,6 +4800,145 @@ ConstraintSystem::simplifyKeyPathApplicationConstraint(
   return unsolved();
 }
 
+Type ConstraintSystem::simplifyAppliedOverloads(
+                                Type fnType,
```

Comment:
Common type computation will eventually put more structure here (e.g., `($T0, $T0) -> $T0`)), which is why I left it as a type (since this will be the signature we expect). I could make it `TypeVariableType` now and generalize later, sure.

Reply:
Yeah, I'll make it `TypeVariableType` for now to be more clear.